ntbs-service/DataMigration/NotificationImportService.cs

```diff
@@ -27,79 +29,150 @@ public class NotificationImportService : INotificationImportService
+                var linkedNotificationId = notification.LegacyId;  
+                _logger.LogInformation(context, requestId, $"Validating notification with Id={linkedNotificationId}");
 
                 var validationErrors = GetValidationErrors(notification);
                 if (validationErrors.Count() == 0)
                 {
-                    _logger.LogInformation(requestId, $"No validation errors found");
+                    _logger.LogInformation(context, requestId, $"No validation errors found");
+                    importResult.AddValidNotification(linkedNotificationId);
                 }
                 else
                 {
                     isAnyNotificationInvalid = true;
-                    _logger.LogWarning(requestId, $"{validationErrors.Count()} validation errors found for notification with Id={linkedNotificationId}:");
+                    importResult.AddValidationErrorsMessages(linkedNotificationId, validationErrors.ToList());
+                    _logger.LogWarning(context, requestId, $"{validationErrors.Count()} validation errors found for notification with Id={linkedNotificationId}:");
                     foreach (var validationError in validationErrors)
                     {
-                        _logger.LogWarning(requestId, validationError.ErrorMessage);
+                        _logger.LogWarning(context, requestId, validationError.ErrorMessage);
                     }
                 }
             }
 
             if (isAnyNotificationInvalid)
             {
-                _logger.LogInformation(requestId, $"Terminating importing notifications for a patient due to validation errors");
-                return null;
+                _logger.LogFailure(context, requestId, $"Terminating importing notifications for {patientName} due to validation errors");
+                return importResult;
             }
 
-            _logger.LogInformation(requestId, $"Importing {notifications.Count()} valid notifications");
+            _logger.LogSuccess(context, requestId, $"Importing {notifications.Count()} valid notifications");
             var savedNotifications = await _notificationImportRepository.AddLinkedNotificationsAsync(notifications);
+            await _migrationRepository.updatedImportedNotificiationsAsync(savedNotifications);
+
 
-            _logger.LogInformation(requestId, $"Finished importing notification for {patientName}");
             var newIdsString = string.Join(" ,", savedNotifications.Select(x => x.NotificationId));
+            _logger.LogSuccess(context, requestId, $"Imported notifications have following Ids: {newIdsString}");
 
-            _logger.LogInformation(requestId, $"Imported notifications have following Ids: {newIdsString}");
-            
-            return savedNotifications;
+            _logger.LogInformation(context, requestId, $"Finished importing notification for {patientName}");
+            return importResult;

```

Comment:
In both cases we are returning the `importResult` variable, so perhaspwe should highlight it with an `if`/`else` structure:
```
if (isAnyNotificationInvalid)
{
  _logger. ...
}
else
{
  ...  // do all the success stuff
}
return importResult;
```